deployment/src/main/java/io/quarkiverse/hivemqclient/deployment/MqttDevServicesProcessor.java

```diff
@@ -0,0 +1,296 @@
+            LoggingSetupBuildItem loggingSetupBuildItem,
+            GlobalDevServicesConfig devServicesConfig) {
+
+        MqttDevServiceCfg configuration = getConfiguration(mqttClientBuildTimeConfig);
+
+        if (devService != null) {
+            boolean shouldShutdownTheBroker = !configuration.equals(cfg);
+            if (!shouldShutdownTheBroker) {
+                return devService.toBuildItem();
+            }
+            shutdownBroker();
+            cfg = null;
+        }
+
+        StartupLogCompressor compressor = new StartupLogCompressor(
+                (launchMode.isTest() ? "(test) " : "") + "MQTT Dev Services Starting:", consoleInstalledBuildItem,
+                loggingSetupBuildItem);
+        try {
+            RunningDevService newDevService = startMqttBroker(dockerStatusBuildItem, configuration, launchMode,
+                    devServicesConfig.timeout);
+            if (newDevService != null) {
+                devService = newDevService;
+
+                Map<String, String> config = devService.getConfig();
+                if (devService.isOwner()) {
+                    log.info("Dev Services for MQTT started.");
+                }
+            }
+            if (devService == null) {
+                compressor.closeAndDumpCaptured();
+            } else {
+                compressor.close();
+            }
+        } catch (Throwable t) {
+            compressor.closeAndDumpCaptured();
+            throw new RuntimeException(t);
+        }
+
+        if (devService == null) {
+            return null;
+        }
+
+        // Configure the watch dog
+        if (first) {
+            first = false;
+            Runnable closeTask = () -> {
+                if (devService != null) {
+                    shutdownBroker();
+
+                    log.info("Dev Services for MQTT shut down.");
+                }
+                first = true;
+                devService = null;
+                cfg = null;
+            };
+            QuarkusClassLoader cl = (QuarkusClassLoader) Thread.currentThread().getContextClassLoader();
+            ((QuarkusClassLoader) cl.parent()).addCloseTask(closeTask);
+        }
+        cfg = configuration;
+        return devService.toBuildItem();
+    }
+
+    private void shutdownBroker() {
+        if (devService != null) {
+            try {
+                devService.close();
+            } catch (Throwable e) {
+                log.error("Failed to stop the MQTT broker", e);
+            } finally {
+                devService = null;
+            }
+        }
+    }
+
+    private RunningDevService startMqttBroker(DockerStatusBuildItem dockerStatusBuildItem,
+            MqttDevServiceCfg config, LaunchModeBuildItem launchMode,
+            Optional<Duration> timeout) {
+        if (!config.devServicesEnabled) {
+            // explicitly disabled
+            log.debug("Not starting Dev Services for MQTT, as it has been disabled in the config.");
+            return null;
```

Comment:
Not sure if applicable here, but I normally see weird returning 'null'. Does it make sense to use Optional in this case?